src/codelabs/async-await/index.md

```diff
@@ -0,0 +1,493 @@
+{% prettify dart %}
+    main() [!async!] {
+{% endprettify %}
+
+If the function has a declared return type, then update the type to be
+`Future<T>`, where `T` is the type of the value that the function returns. 
+If the function doesn't explicitly return a value, then the return type is
+`Future<void>`:
+
+{% prettify dart %}
+    [!Future<void>!] main() async {
+{% endprettify %}
+
+Now that you have an `async` function, you can use the `await` keyword to wait
+for a future to complete:
+{% prettify dart %}
+    print(await createOrderMessage());
+{% endprettify %}
+
+
+As the following two examples show, the `async` and `await` keywords result in 
+asynchronous code that looks a lot like synchronous code. 
+The only differences are highlighted in the asynchronous example, which — if 
+your window is wide enough — is to the right of the synchronous example.
+
+<div class="container">
+  <div class="row">
+    <div class="col-sm">
+    <h4> Example: synchronous functions</h4>
+
+{% prettify dart %}
+// Synchronous
+String createOrderMessage () {
+  var order = getUserOrder();
+  return 'Your order is: $order';
+}
+
+Future<String> getUserOrder() {
+  // Imagine that this function is
+  // more complex and slow.
+  return 
+    Future.delayed(
+      Duration(seconds: 4), () => 'Large Latte'); 
+}
+
+// Synchronous
+main() {
+  print('Fetching user order...');
+  print(createOrderMessage());
+}
+
+// "Your order is: Instance of '_Future<String>'"
+{% endprettify %}
+    </div>
+    <div class="col-sm">
+    <h4> Example: asynchronous functions</h4>
+
+{% prettify dart %}
+// Asynchronous
+[!Future<String>!] createOrderMessage () [!async!] {
+  var order = [!await!] getUserOrder();
+  return 'Your order is: $order';
+}
+
+Future<String> getUserOrder() {
+  // Imagine that this function is
+  // more complex and slow.
+  return
+   Future.delayed(
+     Duration(seconds: 4), () => 'Large Latte');
+}
+
+// Asynchronous
+main() [!async!] {
+  print('Fetching user order...');
+  print([!await!] createOrderMessage());
+}
+
+// "Your order is: Large Latte"
+{% endprettify %}
+    </div>
+  </div>
+</div>
+
+The asynchronous example is different in three ways:
+  * The return type for `createOrderMessage()` changes from `String` to 
+  `Future<String>`.
+  * The **`async`** keyword appears before the function bodies for 
+  `createOrderMessage()` and `main()`.
+  * The **`await`** keyword appears before calling the asynchronous functions 
+  `getUserOrder()` and `createOrderMessage()`.
+
+{{ site.alert.secondary }}
+  **Key terms:**
+* **async**: You can use the `async` keyword before a function's body to mark it as
+asynchronous.
+* **async function**:  An `async` function is a function labeled with the `async` 
+keyword. 
+* **await**: You can use the `await` keyword to get the completed result of an
+asynchronous expression. The `await` keyword only works within an `async` function.
+{{ site.alert.end }}
+
+## Execution flow with async and await
```

Comment:
`##` -> `###`
(I noticed in the side nav that the exercise applies to "Working with futures", not "Execution flow". I think this change fixes that problem.)